### PR TITLE
Add Penman-Monteith and 4 other ET calculation methods

### PR DESCRIPTION
**Pull request by stcui007**
Refactor the ET calculation functions into 5 independent functions

## Changes
models/kernels/EtCalcFunction.hpp
test/models/hymod/include/Et_Calc_Function_Test.cpp 

## Testing
Unit test

## Notes

-

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows project standards (link if applicable)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

## Testing checklist (automated report can be put here)

1.

### Target Environment support

- [ ] Linux

### CODE REVIEW

**Comment by hellkite500** [comment]
Hold on #52 and #59 since this PR changes a file name that those are entangled with.

**Comment by stcui007** [comment]
Anything I do to make it easier?

File: models/kernels/EtCalcFunction.hpp [hunk 1]
```diff
@@ -0,0 +1,1115 @@
...
+//
+// According to the UN FAO Penman-Monteith example here: 
+//        http://www.fao.org/3/X0490E/x0490e06.htm#aerodynamic%20resistance%20(ra)
+// the zero plane roughness length,"d" can be approximated as 2/3 of the vegetation height (H): d=2/3*H.
+// the momentum roughness height "zom" can be estimated as 0.123*H.
+// the heat transfer roughness height "zoh" can be approximated as 0.1 * zom.
+//-----------------------------------------------------------------------------------------------------------------
+
+//DATA STRUCTURE TO HOLD AORC FORCING DATA
+struct aorc_forcing_data
```
Comment (jdmattern-noaa) [inline models/kernels/EtCalcFunction.hpp:78]: There is a struct for this in the forcing class. Do we want one here too?

File: models/kernels/EtCalcFunction.hpp [hunk 2]
```diff
@@ -0,0 +1,1115 @@
...
+{
+// struct NAME                          DESCRIPTION                                            ORIGINAL AORC NAME     
+//____________________________________________________________________________________________________________________
+float precip_kg_per_m2;                // Surface precipitation "kg/m^2"                         | APCP_surface
```
Comment (jdmattern-noaa) [inline models/kernels/EtCalcFunction.hpp:82]: Do we want everything to be doubles instead of floats?

File: models/kernels/EtCalcFunction.hpp [hunk 3]
```diff
@@ -0,0 +1,1115 @@
...
+#define FALSE 0
+
+#define CP  1.006e+03  //  specific heat of air at constant pressure, J/(kg K), a physical constant.
+#define KV2 0.1681     //  von Karman's constant squared, equal to 0.41 squared, unitless
+#define TK  273.15     //  temperature in Kelvin at zero degree Celcius
+#define SB  5.67e-08   //  stefan_boltzmann_constant in units of W/m^2/K^4
+
+// NOTE: SET YOUR EDIT WINDOW TO 120 CHARACTER WIDTH TO READ THIS CODE IN ITS ENTIRETY.
+
+//#####################################
```
Comment (jdmattern-noaa) [inline models/kernels/EtCalcFunction.hpp:15]: I suggest following the Doxygen style comments that Chris added. Example here:
https://github.com/NOAA-OWP/ngen/blob/master/include/geojson/FeatureBuilder.hpp#L23

File: models/kernels/EtCalcFunction.hpp [hunk 4]
```diff
@@ -0,0 +1,1115 @@
+#include <iostream>
```
Comment (jdmattern-noaa) [inline models/kernels/EtCalcFunction.hpp:1]: I recommend breaking this file up into multiple files and classes.

File: models/kernels/schaake_test.cpp [hunk 5]
```diff
@@ -1,134 +0,0 @@
-#include "utility.hpp"
```
Comment (jdmattern-noaa) [inline models/kernels/schaake_test.cpp:1]: I recommend making this like the other test file that incorporates the google test framework.

File: test/models/hymod/include/Et_Calc_Function_Test.cpp [hunk 6]
```diff
@@ -0,0 +1,1179 @@
...
+if(et_options.use_energy_balance_method ==TRUE)   printf("energy balance method:\n");
+if(et_options.use_aerodynamic_method ==TRUE)      printf("aerodynamic method:\n");
+if(et_options.use_combination_method ==TRUE)      printf("combination method:\n");
+if(et_options.use_priestley_taylor_method ==TRUE) printf("Priestley-Taylor method:\n");
+if(et_options.use_penman_monteith_method ==TRUE)  printf("Penman Monteith method:\n");
+                                                 
+//printf("calculated instantaneous potential evapotranspiration (PET) =%8.6e m/s\n",et_m_per_s);
+//printf("calculated instantaneous potential evapotranspiration (PET) =%8.6lf mm/d\n",et_m_per_s*86400.0*1000.0);
+
+//EXPECT_DOUBLE_EQ (0.0, et_m_per_s-1.106268e-07);
```
Comment (jdmattern-noaa) [inline test/models/hymod/include/Et_Calc_Function_Test.cpp:261]: If the EXPECT_DOUBLE_EQ is giving problems with the precision, you can incorporate rounding like here:
https://github.com/NOAA-OWP/ngen/blob/master/test/forcing/Forcing_Test.cpp#L137

**Comment by hellkite500** [comment]
Replaced by #102

**Pull request closed without merging**